setup/main.go

```diff
@@ -0,0 +1,1653 @@
+	Check: func(state *SetupState) (bool, error) {
+		if state.SkipAutomatedExplain {
+			return true, nil
+		}
+		logExplain, err := usingLogExplain(state.CurrentSection)
+		if err != nil || logExplain {
+			return logExplain, err
+		}
+		spl, err := getPendingSharedPreloadLibraries(state.QueryRunner)
+		if err != nil {
+			return false, err
+		}
+		return strings.Contains(spl, "auto_explain"), nil
+	},
+	Run: func(state *SetupState) error {
+		var doAdd bool
+		err := survey.AskOne(&survey.Confirm{
+			Message: "Add auto_explain to shared_preload_libraries (will be saved to Postgres)?",
+			Default: false,
+			Help:    "Postgres will have to be restarted in a later step to apply this configuration change; learn more about shared_preload_libraries here: https://www.postgresql.org/docs/current/runtime-config-client.html#GUC-SHARED-PRELOAD-LIBRARIES",
+		}, &doAdd)
+		if err != nil {
+			return err
+		}
+		if !doAdd {
+			return nil
+		}
+
+		existingSpl, err := getPendingSharedPreloadLibraries(state.QueryRunner)
+		if err != nil {
+			return err
+		}
+		var newSpl string
+		if existingSpl == "" {
+			newSpl = "auto_explain"
+		} else {
+			newSpl = existingSpl + ",auto_explain"
+		}
+		return applyConfigSetting("shared_preload_libraries", newSpl, state.QueryRunner)
+	},
+}
+
+var reloadCollector = &Step{
+	Description: "Reload collector configuration",
+	Check: func(state *SetupState) (bool, error) {
+		// N.B.: there's no way to tell whether the collector actually needs to reload; we
+		// force this once per setup helper invocation (this starts out as false and is set
+		// to true below)
+		return state.DidReload, nil
+	},
+	Run: func(state *SetupState) error {
+		var doReload bool
+		err := survey.AskOne(&survey.Confirm{
+			Message: "The collector configuration must be reloaded for changes to take effect; reload now?",
+			Default: false,
+		}, &doReload)
+		if err != nil {
+			return err
+		}
+		if !doReload {
+			return nil
+		}
+		err = reload()
+		if err != nil {
+			return err
+		}
+		state.DidReload = true
+		return nil
+	},
+}
+
+var restartPg = &Step{
+	Description: "If necessary, restart Postgres to have configuration changes take effect",
+	Check: func(state *SetupState) (bool, error) {
+		row, err := state.QueryRunner.QueryRow("SELECT COUNT(*) FROM pg_settings WHERE pending_restart;")
+		if err != nil {
+			return false, err
+		}
+		return row.GetInt(0) == 0, nil
+	},
+	Run: func(state *SetupState) error {
+		rows, err := state.QueryRunner.Query("SELECT name FROM pg_settings WHERE pending_restart")
+		if err != nil {
+			return err
+		}
+		var pendingSettings []string
+		for _, row := range rows {
+			pendingSettings = append(pendingSettings, row.GetString(0))
+		}
+
+		pendingList := getConjuctionList(pendingSettings)
+		var restartNow bool
+		err = survey.AskOne(&survey.Confirm{
+			Message: fmt.Sprintf("Postgres must be restarted for changes to %s to take effect; restart Postgres now?", pendingList),
```

Comment:
This one is probably the most impactful step as it would be visible to any applications connected - can we somehow give extra attention to this? (use different color, add exclamation mark, etc)